src/web/mod.rs

```diff
@@ -126,6 +127,15 @@ use tower::ServiceBuilder;
 use tower_http::trace::TraceLayer;
 use url::form_urlencoded;
 
+// from https://github.com/servo/rust-url/blob/master/url/src/parser.rs
+// and https://github.com/tokio-rs/axum/blob/main/axum-extra/src/lib.rs
```

Comment:
Are these APIs not exposed publicly? This seems ~fine since the spec isn't going to change, but it would be nice to reuse code if possible.

Reply:
Sadly not publicly. 

But thinking about this, I'll check with axum if this is a piece they could re-export.